Error when using Vulkan backend via `use-ash-v0-38` feature

I made a blank project and it fails to build due to:

```
error[E0605]: non-primitive cast: `i32` as `ash::vk::definitions::SurfaceKHR`
    --> C:\Users\EidLoi\.cargo\registry\src\index.crates.io-6f17d22bba15001f\sdl3-0.12.0\src/sdl3\video.rs:1610:41
     |
1610 |         let mut surface: VkSurfaceKHR = 0 as _;
     |                                         ^^^^^^ an `as` expression can only be used to convert between primitive types or to coerce to a specific trait object
```

`rustc 1.83.0 (90b35a623 2024-11-26)` on Windows/cmake/build from source

Deps:

```toml
[dependencies]
sdl3 = { version = "0.12.0", features = ["build-from-source"] }

[dependencies.sdl3-sys]
version = "0.2.0+SDL3-preview-3.1.8"
features = ["use-ash-v0-38"]
```

--- a/Cargo.toml
+++ b/Cargo.toml
@@ -60,7 +60,7 @@ link-framework = ["sdl3-sys/link-framework"]
 build-from-source = ["sdl3-sys/build-from-source"]
 build-from-source-static = ["sdl3-sys/build-from-source-static"]
 build-from-source-unix-console = ["sdl3-sys/sdl-unix-console-build"]
-
+ash = ["sdl3-sys/use-ash-v0-38"]
 default = []
 unsafe_textures = []
 gfx = ["c_vec"]      #, "sdl3-sys/gfx"]

--- a/src/sdl3/video.rs
+++ b/src/sdl3/video.rs
@@ -1611,6 +1611,10 @@ impl Window {
     /// function in the Vulkan library.
     #[doc(alias = "SDL_Vulkan_CreateSurface")]
     pub fn vulkan_create_surface(&self, instance: VkInstance) -> Result<VkSurfaceKHR, Error> {
+        #[cfg(feature = "ash")]
+        let mut surface: VkSurfaceKHR = VkSurfaceKHR::default();
+
+        #[cfg(not(feature = "ash"))]
         let mut surface: VkSurfaceKHR = 0 as _;
         if unsafe {
             sys::vulkan::SDL_Vulkan_CreateSurface(self.context.raw, instance, null(), &mut surface)

